src/db.rs

```diff
@@ -323,3 +321,462 @@ pub async fn dependencies_by_dataset_select(
 
     Ok(dependencies)
 }
+
+#[cfg(test)]
```

Comment:
@coderabbitai Please create the following unit test for the following functions in the `db.rs` file: 
- data_product_by_dataset_select
  - Do we get all data products for a dataset?
- dependency_upsert
  - Can we add a dependence?
  - Can we update a dependence?
  - Do we get rejected if the dependency parent refers to a non-existent data product?
  - Do we get rejected if the dependency child refers to a non-existent data product?
  - Do we get rejected if a dependency has the same data product for both the parent and child?
- depencencie_by_dataset_select
  - Do we get all dependencies for a dataset?

Please create a PR for these changes.